bundle/libraries/same_name_libraries.go

```diff
@@ -0,0 +1,104 @@
+package libraries
+
+import (
+	"context"
+	"path/filepath"
+	"strings"
+
+	"github.com/databricks/cli/bundle"
+	"github.com/databricks/cli/libs/diag"
+	"github.com/databricks/cli/libs/dyn"
+)
+
+type checkForSameNameLibraries struct{}
+
+var patterns = []dyn.Pattern{
+	taskLibrariesPattern.Append(dyn.AnyIndex(), dyn.AnyKey()),
+	forEachTaskLibrariesPattern.Append(dyn.AnyIndex(), dyn.AnyKey()),
+	envDepsPattern.Append(dyn.AnyIndex()),
```

Comment:
Q: this matches on any key, but we know it only works for "whl" and "jar" (and maybe more).

Could the pattern be more specific to reflect this?

Reply:
Good point, followed up here: https://github.com/databricks/cli/pull/2401